hack/build.sh

```diff
@@ -16,8 +16,9 @@ fi
 MODE="${MODE:-release}"
 GIT_COMMIT="${SOURCE_GIT_COMMIT:-$(git rev-parse --verify 'HEAD^{commit}')}"
 GIT_TAG="${BUILD_VERSION:-$(git describe --always --abbrev=40 --dirty)}"
+HOSTARCH="$(go env GOHOSTARCH)"
```

Comment:
ok so iiuc - the architecture is now set to the host architecture if built manually and set to the architecture of the payload if it is extracted from the payload? so if i actually custom build the installer on an ARM laptop it would set the architecture in the install-config to arm64 - but which might not match the payload architecture necessarily.

Also note that someone could still override the payload using OPENSHIFT_RELEASE_IMAGE_OVERRIDE. so if these are called out it should make things clear for upstream as well as downstream.